config/defaults.cmake

```diff
@@ -218,6 +220,9 @@ set(BLAS_LIBRARIES "" CACHE STRING "The BLAS library.")
 set(LAPACK_INCLUDE_DIRS "" CACHE STRING "Path to LAPACK headers.")
 set(LAPACK_LIBRARIES "" CACHE STRING "The LAPACK library.")
 
+set(CODIPACK_INCLUDE_DIRS "${MFEM_DIR}/../CoDiPack/inlude" CACHE STRING "Path to CoDoPack headers.")
```

Comment:
typo -> `s/CoDoPack/CoDiPack`

Reply:
The defaults.cmake is fixed. 

In forward mode (MFEM_USE_ADFORWARD) the Hessian is computed by applying forward differentiation twice. For reverse mode we have to use forward for one of the steps, we cannot use reverse mode twice. That is the natural way of employing AD in reverse mode for high-order derivatives.